src/main/scala/mesosphere/marathon/state/ResourceRole.scala

```diff
@@ -8,6 +8,14 @@ import mesosphere.marathon.api.v2.Validation._
 object ResourceRole {
   val Unreserved = "*"
 
+  def validForRole(validRoles: Set[String]): Validator[Set[String]] = {
+    isTrue(s"""acceptedResourceRoles must be either [*] or one of ${validRoles.mkString("[", ",", "]")}""") { acceptedResourceRoles =>
+      acceptedResourceRoles.isEmpty ||
+        (acceptedResourceRoles.size == 1 &&
+          (acceptedResourceRoles.head.equals(Unreserved) || validRoles.contains(acceptedResourceRoles.head)))
```

Comment:
I'm a little confused. is `acceptedResourceRoles` only allowed to be empty or size one? Why is that?

Reply:
After rereading the specs, i'm not totally sure. It seems that acceptedResourceRoles are actually not really compatible with custom roles? @timcharper, any clarifications?

Reply:
I'm still missing something here. Assuming we have a runSpec with `acceptedResourceRoles: ["slave_public"]` and marathon starts with `--mesos_role *` (or without --mesos_role).
After the migration for apps and pods is done, the runSpec would have `role: "*"`, which would then be invalid for `acceptedResourceRoles: ["slave_public"]`

Do we allow any acceptedResource roles if the role is set to "*"?

Reply:
If `--mesos_role` is `*`, then acceptedResourceRoles can only be `"*"`

Reply:
(I sure hope this is validated... lol)

Reply:
Hmmm. Then the default for `--mesos_role` should probably be 'slave_public', not '*', right? Otherwise I would assume we would break a lot of existing setups, as usually marathon is started without `--mesos_role` and users are setting `acceptedResourceRoles` to `slave_public` to assign the service to public facing nodes?

Reply:
This is **not** validated.

Reply:
Ok, i've updated the acceptedResourceRoleValidation to only accept the runSpec.role and/or *

Reply:
Great, this is the correct behavior.

And.... shoot, it's not validated right now. Let's go forward with it validated for now. We'll likely need to implement a feature to fix nosensical resource roles, on-they-fly, and then have a deprecated-feature option to automatically fix them when they're posted erroneously